File: boardinghouse_backend/app/Models/BoardingHouseType.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

class BoardingHouseType extends Model
{
    use HasFactory;

    protected $fillable = [
        'type',
        'name',
        'slug',
        'description',
        'is_visible',
        'position',
        'sort'
    ];
    protected $casts = [
        'is_visible' => 'boolean',
    ];

    public function boarding_houses(): HasMany
    {
        return $this->hasMany(BoardingHouse::class, 'boarding_house_type_id');
    }
}

File: boardinghouse_app/lib/models/boarding_house_type.dart
class BoardingHouseType {
  int? id;
  String? name;
  // String? featuredImageId;
  // String? slug;
  // String? description;
  // int? position;
  // bool? isVisible;
  // int? sort;

  BoardingHouseType({
    this.id,
    this.name,
    // this.featuredImageId,
    // this.slug,
    // this.description,
    // this.position,
    // this.isVisible,
    // this.sort,
  });

  factory BoardingHouseType.fromJson(Map<String, dynamic> json) {
    return BoardingHouseType(
      id: json['id'],
      name: json['name'],
      // featuredImageId: json['featured_image_id'],
      // slug: json['slug'],
      // description: json['description'],
      // position: json['position'],
      // isVisible: json['is_visible'],
      // sort: json['sort'],
    );
  }

  // Map<String, dynamic> toJson() {
  //   return {
  //     'id': id,
  //     'name': name,
  //     'featured_image_id': featuredImageId,
  //     'slug': slug,
  //     'description': description,
  //     'position': position,
  //     'is_visible': isVisible,
  //     'sort': sort,
  //   };
  // }
}

